backend/src/test/java/codezap/tag/service/TemplateTagServiceTest.java

```diff
@@ -0,0 +1,271 @@
+
+import java.util.Arrays;
+import java.util.List;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.transaction.annotation.Transactional;
+
+import codezap.category.domain.Category;
+import codezap.fixture.CategoryFixture;
+import codezap.fixture.MemberFixture;
+import codezap.fixture.TemplateFixture;
+import codezap.global.ServiceTest;
+import codezap.member.domain.Member;
+import codezap.tag.domain.Tag;
+import codezap.tag.dto.response.FindAllTagsResponse;
+import codezap.tag.dto.response.FindTagResponse;
+import codezap.template.domain.Template;
+import codezap.template.domain.TemplateTag;
+
+
+@Transactional
+class TemplateTagServiceTest extends ServiceTest {
+
+    @Autowired
+    private TemplateTagService templateTagService;
+
+    @Nested
+    @DisplayName("템플릿에 대한 태그 생성")
+    class CreateTags {
+
+        @Test
+        @DisplayName("성공: 템플릿에 해당하는 태그 생성")
+        void createTags() {
+            // given
+            Template template = createTemplate();
+            List<String> tagNames = Arrays.asList("tag1", "tag2", "tag3");
+
+            // when
+            templateTagService.createTags(template, tagNames);
+
+            // then
+            List<String> savedTemplateTagNames = getSavedTagNames(template);
+            assertThat(savedTemplateTagNames).containsExactlyElementsOf(tagNames);
+        }
+
+        @Test
+        @DisplayName("성공: 이미 있는 템플릿 태그가 포함된 경우 중복 저장하지 않고 새 태그만 생성")
+        void createTags_WhenExistTemplateTagContains() {
+            // given
+            Template template = createTemplate();
+            Tag existTag = tagRepository.save(new Tag("tag1"));
+            TemplateTag existTemplateTag = new TemplateTag(template, existTag);
+            List<String> tagNames = Arrays.asList(existTag.getName(), "tag2", "tag3");
+
+            // when
+            templateTagService.createTags(template, tagNames);
+
+            // then
+            List<String> savedTemplateTagNames = getSavedTagNames(template);
+            assertThat(savedTemplateTagNames).containsExactlyElementsOf(tagNames);
+        }
+
+        @Test
+        @DisplayName("성공: 이미 있는 태그이지만 이 템플릿의 태그가 아닌 경우 템플릿 태그만 추가")
+        void createTags_WhenExistTagContains() {
+            // given
+            Template template = createTemplate();
+            Tag existTag = tagRepository.save(new Tag("tag1"));
+            List<String> tagNames = Arrays.asList(existTag.getName(), "tag2", "tag3");
+            List<String> beforeTemplateTags = getSavedTagNames(template);
+
+            // when
+            templateTagService.createTags(template, tagNames);
+
+            // then
+            List<String> afterTemplateTags = getSavedTagNames(template);
+            assertAll(
+                    () -> assertThat(beforeTemplateTags).doesNotContain(existTag.getName()),
```

Comment:
저장 전에는 tag2, tag3도 포함되어 있지 않다는 것도 추가되면 좋을 것 같아요!!
```suggestion
                    () -> assertThat(beforeTemplateTags).doesNotContainAnyElementsOf(tagNames),
```